src/app/server/Server.h

```diff
@@ -244,30 +245,31 @@ struct CommonCaseDeviceServerInitParams : public ServerInitParams
         {
             // WARNING: PersistentStorageOpCertStore::Finish() is never called. It's fine for
             //          for examples and for now, since all storage is immediate for that impl.
-            ReturnErrorOnFailure(sPersistentStorageOpCertStore.Init(this->persistentStorageDelegate));
-            this->opCertStore = &sPersistentStorageOpCertStore;
+            ReturnErrorOnFailure(sPersistentStorageOpCertStore->Init(this->persistentStorageDelegate));
+            this->opCertStore = &sPersistentStorageOpCertStore.get();
         }
 
         // Injection of report scheduler WILL lead to two schedulers being allocated. As recommended above, this should only be used
         // for IN-TREE examples. If a default scheduler is desired, the basic ServerInitParams should be used by the application and
         // CommonCaseDeviceServerInitParams should not be allocated.
         if (this->reportScheduler == nullptr)
         {
-            reportScheduler = &sReportScheduler;
+            sReportScheduler->Init(&sTimerDelegate.get());
```

Comment:
This will not get called on injection of a report scheduler, we can either keep the nullptr check in the constructor that takes a TimerDelegate as a param, or have an `EnsureInit` that takes no parameter and that gets called in an  `else` condition here.

Reply:
I'm not sure I get your point. Why initialise that global at all, especially when it's not going to be used? The whole point of this PR is to allow us to not even allocate the memory for it.

Reply:
`this->reportScheduler == nullptr` only checks if there were no report Scheduler injected. If this check fails, that means the reportScheduler was injected somewhere else, so why not check that it is initialized?

something like:
`reportScheduler.EnsureInit()` to make sure that whatever is injected has been properly generated?

Reply:
Doesn't the same logic apply to all other things that can be injected? My impression was you need to Init() them before injecting (in this case you can alternatively construct it with a pointer). I don't see EnsureInit() used for any of the other ones...

Reply:
As noted above, I think we should avoid changing how ReportScheduler is initialized.